common-custom-user-data-maven-extension/src/main/java/com/gradle/ApiAccessor.java

```diff
@@ -43,19 +39,18 @@ private static void ensureBuildScanApiIsAccessible(Class<?> extensionClass) thro
         }
     }
 
-    private static BuildScanApi lookupBuildScanApi(PlexusContainer container) throws MavenExecutionException {
-        if (!container.hasComponent(BUILD_SCAN_API_CONTAINER_OBJECT)) {
+    private static <T> T lookupClass(Class<T> componentClass, String component, PlexusContainer container) throws MavenExecutionException {
+        if (!container.hasComponent(component)) {
```

Comment:
❓ Can we use `PlexusContainer.hasComponent(Class)` and `PlexusContainer.lookup(Class)` here? That would allow us to get rid of the `component` String.

Reply:
Done

Reply:
This could use the class now again.

Reply:
yep fixed.